tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -115,61 +122,57 @@ kbs_set_resource_from_file() {
 	path+="${type}/"
 	path+="${tag}"
 
-	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
-		--auth-private-key "$KBS_PRIVATE_KEY" set-resource \
-		--path "$path" --resource-file "$file"
+	docker run \
+		-v "$path:$path" \
+		-v "$file:$file" \
+		--privileged \
+		--net=host \
+			ghcr.io/kata-containers/kata-containers/coco-kbs-client:${cli_features}-${rust_version}-${trustee_version}-${uname -m} -- \
+				kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
+				--auth-private-key "$KBS_PRIVATE_KEY" set-resource \
+				--path "$path" --resource-file "$file"
 }
 
-# Build and install the kbs-client binary, unless it is already present.
-#
-kbs_install_cli() {
-	command -v kbs-client >/dev/null && return
-
-	source /etc/os-release || source /usr/lib/os-release
-	case "${ID}" in
-		ubuntu)
-			local pkgs="build-essential"
-
-			sudo apt-get update -y
-			# shellcheck disable=2086
-			sudo apt-get install -y $pkgs
-			;;
-		centos)
-			local pkgs="make"
-
-			# shellcheck disable=2086
-			sudo dnf install -y $pkgs
-			;;
-		*)
-			>&2 echo "ERROR: running on unsupported distro"
-			return 1
-			;;
-	esac
+kbs_get_cli_container_image_name() {
+	ensure_yq
 
 	# Mininum required version to build the client (read from versions.yaml)
-	local rust_version
-	ensure_yq
 	rust_version=$(get_from_kata_deps "externals.coco-trustee.toolchain")
-	# Currently kata version from version.yaml is 1.72.0
-	# which doesn't match the requirement, so let's pass
-	# the required version.
-	_ensure_rust "$rust_version"
-
-	pushd "${COCO_KBS_DIR}"
-	# Compile with sample features to bypass attestation.
-	make CLI_FEATURES=sample_only cli
-	sudo make install-cli
-	popd
+
+	# trustee version (read from versions.yaml)
+	trustee_version=$(get_from_kata_deps "externals.coco-trustee.version")
+
+	# The container image name
+	coco_kbs_client_container_image="ghcr.io/kata-containers/kata-containers/coco-kbs-client:${KBS_CLI_FEATURES}-${rust_version}-${trustee_version}-${uname -m}"
+
+	echo "${coco_kbs_client_container_image}"
+}
+
+kbs_build_cli_container() {
+	coco_kbs_client_container_image=$(kbs_get_cli_container_image_name)
+
+	ensure_yq
+	docker build \
+		--build-arg RUST_TOOLCHAIN="$(get_from_kata_deps "externals.coco-trustee.toolchain")" \
+		--build-arg TRUSTEE_VERSION="$(get_from_kata_deps "externals.coco-trustee.version")" \
+		--build-arg CLI_FEATURES="${KBS_CLI_FEATURES}" \
+		-t ${coco_kbs_client_container_image}
+		-f Dockerfile.kbs-client .
+
+	docker push ${coco_kbs_client_container_image}
+}
+
+# If a docker image is not existent for the kbs-client, build it
+kbs_install_cli() {
+	# The name of this function is misleading ...
+
+	# but it's left as it is for backward compatibility with the CI
```

Comment:
There looks to be indirection with the CI workflow via https://github.com/kata-containers/kata-containers/blob/7886ed66703221d082aa4b03a6f73ccf0da39458/tests/integration/kubernetes/gha-run.sh#L268-L270, so I think we could rename this function and update it in gha-run without breaking anything?

Reply:
Not really, the full rename will come in a separate PR, as the function is called from run-kata-coco-tests.yaml

Reply:
I don't understand. `run-kata-coco-tests.yaml` calls `tests/integration/kubernetes/gha-run.sh install-kbs-client`, which calls `install_kbs_client` as I linked, so renaming `kbs_install_cli` here and then updating `install_kbs_client` won't break anything?

Reply:
Ah, that won't break anything, indeed.  However, install-kbs-client is still a misleading name (as kbs_install_cli is as well).

I was planning to update those all together as part of a follow-up PR, as those changes mostly won't require CI running.

Reply:
I changed the name as part of this PR, leaving only the yaml change for a follow-up one.

Reply:
Cool. Thanks!